### ISSUE TITLE
Deleting doesn't remove data in database

### ISSUE THREAD

**Issue by oscar-b** [jrbasso/MeioUpload#34]
When deleting, _unsetDataFields is called. This function however, doesn't remove (replace the existing data with nothing), instead it removes the fields from the data array being saved. The only thing that leads to is that nothing happens to the fields, since there is no data to overwrite the existing data with.

The Wiki says that:
"The uploaded file including possible thumbnails will then be deleted together with adherent database fields upon save."

/Oscar

**Comment by oscar-b** [comment]
I think this is caused by this commit:
http://github.com/jrbasso/MeioUpload/commit/37c5ba6cb6201ef5852a7562553bdd3696f06623#L1R637

As the comment says, deletion is handled further down.

**Comment by TheRec** [comment]
I think so too. I would add that the use of "_unsetDataFields" is what could be the source of confusion. This function will unset the array keys and thus in an update situation it will just not touch the present data. The function should not be modified though because it is used in the case of the no-db mode. What I suggest would be to use another function to nullify the data fields, like this :

```
/**
 * Sets as NULL data from $data
 * Useful when deleting a file.
 *
 * @param $modelName string name of the Model
 * @param $fieldName string name of field that holds a reference to the file
 * @param $data array
 * @param $options array
 * @return array
 * @author Jose Diaz-Gonzalez
 **/
  function _nullifyDataFields($modelName, $fieldName, $data, $options) {
    if (isset($data[$modelName][$fieldName])) {
      $data[$modelName][$fieldName] = null;
    }
    if (isset($data[$modelName][$options['fields']['dir']])) {
      $data[$modelName][$options['fields']['dir']] = null;
    }
    if (isset($data[$modelName][$options['fields']['filesize']])) {
      $data[$modelName][$options['fields']['filesize']] = null;
    }
    if (isset($data[$modelName][$options['fields']['mimetype']])) {
      $data[$modelName][$options['fields']['mimetype']] = null;
    }
    return $data;
  }
```

**Comment by jrbasso** [comment]
Nullifing fields on remove. Closed by 01ffbbdbd054b88cf439d3ecb6cf72fd9b2f16b7
